src/services/DfusionService.ts

```diff
@@ -246,13 +256,16 @@ export class DfusionRepoImpl implements DfusionService {
     let token = this._tokenCache[tokenAddress]
 
     if (!token) {
-      const networkId = await this._getNetworkId()
-
       const tokenContract = this._erc20Contract.clone()
       tokenContract.options.address = tokenAddress
 
-      // Get basic data from the contract
-      const { symbol, decimals, name } = await _getDataFromErc20(tokenContract)
+      const [networkId, { symbol, decimals, name }, tcr] = await Promise.all([
+        this._getNetworkId(),
+        // Get basic data from the contract
+        _getDataFromErc20(tokenContract),
```

Comment:
I'm more and more concerned how much requests we make on each event, requests that can be perfectly cached, like Token data.
I've already run into stuck promises when processing a lot of events when testing event buffering before.
Not for this PR but we should address caching.

Reply:
You are right, I didn't look what function this was in :see_no_evil:

Reply:
> Maybe the TCR should have different cache than the Token info. Token info can be cached forever

Yes, I'm caching the TCR for 5 min, as opposed to Token info that's cached forever.